db/blob/blob_file_builder_test.cc

```diff
@@ -0,0 +1,574 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#include "db/blob/blob_file_builder.h"
+
+#include <cassert>
+#include <cinttypes>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "db/blob/blob_file_addition.h"
+#include "db/blob/blob_index.h"
+#include "db/blob/blob_log_format.h"
+#include "db/blob/blob_log_reader.h"
+#include "env/composite_env_wrapper.h"
+#include "env/mock_env.h"
+#include "file/filename.h"
+#include "file/random_access_file_reader.h"
+#include "options/cf_options.h"
+#include "rocksdb/env.h"
+#include "rocksdb/file_checksum.h"
+#include "rocksdb/options.h"
+#include "test_util/sync_point.h"
+#include "test_util/testharness.h"
+#include "util/compression.h"
+#include "utilities/fault_injection_env.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+class TestFileNumberGenerator {
+ public:
+  uint64_t operator()() { return ++next_file_number_; }
+
+ private:
+  uint64_t next_file_number_ = 1;
+};
+
+class BlobFileBuilderTest : public testing::Test {
+ protected:
+  BlobFileBuilderTest() : env_(Env::Default()), fs_(&env_) {}
```

Comment:
Nit: maybe we can use `Env::LoadEnv` and a env var to create the right type of `Env`, similar to `db_basic_test.cc`. This will allow us to test blob db on other FS.

Reply:
Hm, `env_` is really a `MockEnv` here so no real file operations should be involved... I'm wondering what (if anything) `MockEnv` uses its target for though?

Reply:
Will rename the member variable to `mock_env_` to make it clear that it's not a real `Env`.